app/src/main/java/com/github/jinsen47/pokefaker/app/DirectionLayout.java

```diff
@@ -1,70 +1,72 @@
 package com.github.jinsen47.pokefaker.app;
 
 import android.content.Context;
+import android.net.Uri;
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.widget.ImageButton;
+import android.widget.ImageView;
 import android.widget.RelativeLayout;
+import android.widget.TextView;
 
+import com.gcssloop.widget.RockerView;
 import com.github.jinsen47.pokefaker.R;
 
 /**
  * Created by Jinsen on 16/7/13.
  */
-public class DirectionLayout extends RelativeLayout implements View.OnClickListener{
+public class DirectionLayout extends RelativeLayout {
+    private Context mContext;
     private View mContentView;
-    private ImageButton mLeft;
-    private ImageButton mRight;
-    private ImageButton mUp;
-    private ImageButton mDown;
+    private ServiceListener mSerLisrener;
+    private RockerView rocker;
     private onDirectionListener mListener;
+    private double agle;
 
-    public DirectionLayout(Context context) {
+    public DirectionLayout(Context context, ServiceListener listener) {
         super(context);
-        init(context);
+        mContext = context;
+        mSerLisrener = listener;
+        init();
     }
 
-    private void init(Context context) {
-        LayoutInflater inflater = LayoutInflater.from(context);
+    private void init() {
+        LayoutInflater inflater = LayoutInflater.from(mContext);
         mContentView = inflater.inflate(R.layout.layout_direction, this);
 
-        mLeft = ((ImageButton) mContentView.findViewById(R.id.left));
-        mRight = ((ImageButton) mContentView.findViewById(R.id.right));
-        mUp = ((ImageButton) mContentView.findViewById(R.id.up));
-        mDown = ((ImageButton) mContentView.findViewById(R.id.down));
+        rocker = (RockerView) mContentView.findViewById(R.id.rocker);
```

Comment:
I dont think the rocker view need a close button, 'cause a flying man dont need to remove his wings.